kit/src/elements/textarea/mod.rs

```diff
@@ -100,12 +107,25 @@ pub fn Input<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
         let _ = eval(&update_char_counter_script);
     }
 
-    let cv2 = current_val.clone();
-    let cv3 = current_val.clone();
+    let eval = use_eval(cx);
+    let cursor_eval = include_str!("./caret_position.js")
+        .replace("$ID", &id2);
+
+    let text_value = use_ref(cx, || value.clone());
+    use_future(cx, value, |val|{
+        to_owned![text_value];
+        async move {
+            *text_value.write_silent() = val;
+        }
+    });
+
+    let do_cursor_update = oncursor_update.is_some();
 
-    let text_value = Rc::new(RefCell::new(value.to_string()));
-    let text_value_onchange = Rc::clone(&text_value);
-    let text_value_onreturn = Rc::clone(&text_value);
+    if let Some(val) = cursor_position.write_silent().take() {
```

Comment:
instead of writing to `cursor_position`, re-rendering the textarea, and checking if `cursor_position` is `Some`, you could use a `use_coroutine` and send a command (with the cursor position) over the channel. This approach would require a hook (a `UseRef<bool>`) to ensure one command isn't submitted until the previous one has finished. One would set the hook to `true` before sending the command and the `use_coroutine` would set it back to `false` after the callback.

Reply:
i dont see how though? that would require `oncursor_update` EventHandler to be send async which is not really possible?

Reply:
use_coroutine returns a channel that can be used from a synchronous callback. the coroutine can execute async code.

Reply:
i found that dioxus doesn't allow calling callbacks from an async context. it's an unfortunate limitation.